Linux x86 cross-compilation fails because of _mm_pause call

*Credits to @heinrich5991 and @LordSk for quickly troubleshooting this issue.*

So compiling with `arch=x86` fails on current master:

```
[1/2] [1] c++ src/engine/client/client.cpp
In file included from /usr/lib/gcc/i586-linux-gnu/4.9/include/immintrin.h:29:0,
                 from src/engine/client/client.cpp:5:
/usr/lib/gcc/i586-linux-gnu/4.9/include/xmmintrin.h: In member function 'bool CClient::LimitFps()':
/usr/lib/gcc/i586-linux-gnu/4.9/include/xmmintrin.h:1238:1: error: inlining failed in call to always_inline 'void _mm_pause()': target specific option mismatch
 _mm_pause (void)
 ^
src/engine/client/client.cpp:1779:15: error: called from here
    _mm_pause();
               ^
```

This is what intel tells us about `_mm_pause`: https://software.intel.com/en-us/node/524249

A simple fix to this is to add the option `-msse2` in bam for x86 architectures.

**Edit:** I also had to hardcode the path to freetype for some reason

--- a/bam.lua
+++ b/bam.lua
@@ -169,6 +169,7 @@ end
 
 function GenerateLinuxSettings(settings, conf, arch, compiler)
 	if arch == "x86" then
+		settings.cc.flags:Add("-msse2") -- for the _mm_pause call
 		settings.cc.flags:Add("-m32")
 		settings.link.flags:Add("-m32")
 	elseif arch == "x86_64" then

